lib/galaxy/workflow/run.py

```diff
@@ -165,8 +166,9 @@ def invoke( self ):
                     # https://github.com/galaxyproject/galaxy/issues/2259
                     if job:
                         workflow_invocation_step.job_id = job.id
-            except modules.DelayedWorkflowEvaluation:
+            except modules.DelayedWorkflowEvaluation as de:
                 step_delayed = delayed_steps = True
+                more_info = "(%s)" % de.why
                 self.progress.mark_step_outputs_delayed( step )
```

Comment:
I think you can be more consistent and also simplify lines 181-183 if:
- you modify line 350 as per my suggestion there
- drop `more_info`
- pass `de.why` to self.progress.mark_step_outputs_delayed()` here